### ISSUE TITLE
Changing integration manager results in odd behaviour when accessing widgets

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-web#11057]
Reported by Valere: if you change your integration manager in Riot Web (from, say, scalar.vector.im to dimension.t2bot.io) then you will still be able to use scalar-provisioned widgets _until you refresh_. After which time, opening a scalar-provisioned widget will render as an unstyled `Forbidden`.

**Comment by BillCarsonFr** [comment]
I think i have step to reproduce.
On riot-web, firefox.
1) Go to settings and switch to dimension.t2bot.io as integration manager
2) Go to a room (with sufficient power level), then tap on the integrations button
3) Wait until it's loaded and integrations are visible

4) Reload the browser tab
5)  Go to settings and switch to scalar-staging.vector.im as integration manager
6) Go to a room (with sufficient power level), then tap on the integrations button
7) IM take times to load, and finally displays a message saying power levels are insufficiant

**Comment by turt2live** [comment]
> Side note, it looks like a scalar token is sent to Dimension (can see it in debugger console)

A scalar token is just an authentication token. It's name is not scoped to Scalar.

----

As for the rest of this: a rageshake would be good to see what each integration manager is doing (both are fairly verbose in the logs).

**Comment by turt2live** [comment]
I have a theory on this that (at least) Riot is not switching `ScalarMessaging` over so it is not prepared to interact with the new manager. On refresh the reason things don't work is a long-standing issue about auth and integration managers that we should be resolving soon.

**Comment by turt2live** [comment]
The conclusion of the issue is actually more subtle: in a privacy-protecting browser, cookies won't be saved. Scalar ends up relying on cookies to ensure that a token can be presented even when the token isn't given to it. This is where the problem comes into play: a widget using scalar.vector.im will normally set a cookie so that even when you use dimension.t2bot.io (or whatever) it'll still connect fine and avoid the Forbidden error. This is problematic when the browser evicts the cookie because Scalar now has no token to use.

The solution was supposed to be continuing to use the widget whitelist to determine if a scalar widget is present, therefore invoking the ToS flow on the widget with scalar.vector.im (and associates). However, making Riot do that assumption in a safe way is ugly af, so would rather not do it. Additionally, having Riot actually support the assumption becomes a major refactoring of integration managers and widgets.

So in short, this issue is blocked until we figure out what an integration manager is supposed to do and how it works.

**Comment by turt2live** [comment]
We believe that the issues here will be fixed by https://github.com/vector-im/riot-web/issues/11354 (or rather, the story that references).
